backend/de.metas.business/src/main/java/de/metas/shipping/api/impl/ShipperTransportationDAO.java

```diff
@@ -145,13 +150,59 @@ public Optional<ShipperTransportationId> getSingleByQuery(@NonNull final Shipper
 
 	private IQuery<I_M_ShipperTransportation> toSqlQuery(final @NonNull ShipperTransportationQuery query)
 	{
-		return queryBL.createQueryBuilder(I_M_ShipperTransportation.class)
-				.addOnlyActiveRecordsFilter()
-				.addEqualsFilter(I_M_ShipperTransportation.COLUMNNAME_M_Shipper_ID, query.getShipperId())
-				.addEqualsFilter(I_M_ShipperTransportation.COLUMNNAME_Shipper_BPartner_ID, query.getShipperBPartnerAndLocationId().getBpartnerId())
-				.addEqualsFilter(I_M_ShipperTransportation.COLUMNNAME_Shipper_Location_ID, query.getShipperBPartnerAndLocationId().getRepoId())
-				.addEqualsFilter(I_M_ShipperTransportation.COLUMNNAME_DateDoc, TimeUtil.asTimestamp(query.getShipDate()))
-				.create();
+		if (query.isAny())
+		{
+			return queryBL.createQueryBuilder(I_M_ShipperTransportation.class)
```

Comment:
IMHO, you shall fail in this case, 
because the "logical alternative" would be to return the entire database.

e.g. de.metas.picking.job_schedule.repository.PickingJobScheduleRepository#toSqlQuery